framework/src/service/ServiceListeners.cpp

```diff
@@ -38,33 +38,44 @@ US_MSVC_PUSH_DISABLE_WARNING(4180) // qualifier applied to function type has no
 
 namespace cppmicroservices {
 
-struct BundleListenerCompare : std::binary_function<std::pair<BundleListener, void*>,
-                                                    std::pair<BundleListener, void*>, bool>
+/**
+ * Predicate functions used for comparison when adding and removing listeners.
+ * The functions ending in "Name" are legacy functions to support backwards compatibility.
+ * These will be removed when we implement functionality to remove listeners only via
+ * their listener tokens.
+ */
+auto BundleListenerCompareName = [](const BundleListener& l1, void* d1,
```

Comment:
I think the function name is misleading. It is comparing function addresses not names.